File: duplicate-assert/python-unittest/DeclarationOrderCheckTest_testParents.py
import unittest

class DeclarationOrderCheckTest(unittest.TestCase):

    def test_parents(self):
        parent = DetailAstImpl()
        parent.set_type(TokenTypes.STATIC_INIT)
        method = DetailAstImpl()
        method.set_type(TokenTypes.METHOD_DEF)
        parent.set_first_child(method)
        ctor = DetailAstImpl()
        ctor.set_type(TokenTypes.CTOR_DEF)
        method.set_next_sibling(ctor)

        check = DeclarationOrderCheck()

        check.visit_token(method)
        messages1 = check.get_messages()

        self.assertEqual(0, len(messages1), "No exception messages expected")

        check.visit_token(ctor)
        messages2 = check.get_messages()

        self.assertEqual(0, len(messages2), "No exception messages expected")

File: duplicate-assert/java-junit/DeclarationOrderCheckTest_testParents.java
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.Test;

public class DeclarationOrderCheckTest extends AbstractModuleTestSupport {

    @Test
    public void testParents() {
        final DetailAstImpl parent = new DetailAstImpl();
        parent.setType(TokenTypes.STATIC_INIT);
        final DetailAstImpl method = new DetailAstImpl();
        method.setType(TokenTypes.METHOD_DEF);
        parent.setFirstChild(method);
        final DetailAstImpl ctor = new DetailAstImpl();
        ctor.setType(TokenTypes.CTOR_DEF);
        method.setNextSibling(ctor);

        final DeclarationOrderCheck check = new DeclarationOrderCheck();

        check.visitToken(method);
        final SortedSet<LocalizedMessage> messages1 = check.getMessages();

        assertEquals(0, messages1.size(), "No exception messages expected");

        check.visitToken(ctor);
        final SortedSet<LocalizedMessage> messages2 = check.getMessages();

        assertEquals(0, messages2.size(), "No exception messages expected");
    }
}
